pkg/didcomm/protocol/didexchange/event.go

```diff
@@ -14,3 +14,30 @@ type Event interface {
 	// invitation ID
 	InvitationID() string
 }
+
+// didExchangeEvent implements didexchange.Event interface.
+type didExchangeEvent struct {
+	connectionID string
+	invitationID string
+}
+
+// ConnectionID returns DIDExchange connectionID.
+func (ex *didExchangeEvent) ConnectionID() string {
+	return ex.connectionID
+}
+
+// InvitationID returns DIDExchange invitationID.
+func (ex *didExchangeEvent) InvitationID() string {
+	return ex.invitationID
+}
+
+// didExchangeEvent for sending events with processing error.
+type didExchangeEventError struct {
+	*didExchangeEvent
+	err error
+}
+
+// Error implements error interface.
+func (ex *didExchangeEventError) Error() string {
+	return ex.err.Error()
```

Comment:
probability of panic, is this possible it can be created with nil error?

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/compare/abea54bccf94c6f84b362bc1673c2639e2ecfa9c..05e7cc5e7e99a90abc09704d05394f0fb8ea61ba#diff-79de382a052bbdb4213b9dd373b0e2f6R42).